consumer/consumererror/README.md

```diff
@@ -0,0 +1,115 @@
+# Consumer errors
+
+This package contains error types that should be returned by a consumer when an
+error occurs while processing telemetry. The error types included in this
+package provide functionality for communicating details about
+the error for use upstream in the pipeline. Ideally the error returned by a
+component in its `consume` function should be from this package.
+
+## Use cases
+
+**Retry logic**: Errors should be allowed to embed rejected telemetry items to
+be retried by the consumer.
+
+**Indicating partial success**: Errors can indicate that not all items were
+accepted, for example as in an OTLP partial success message.
+
+**Communicating network error codes**: Errors should allow embedding information
+necessary for the Collector to act as a proxy for a backend, i.e. relay a status
+code returned from a backend in a response to a system upstream from the
+Collector.
+
+## Creating Errors
+
+Errors can be created through one of the three constructors provided:
+
+- `consumererror.New[Signal]` for retryable errors.
+- `consumererror.NewPartial` for errors where only some of the items failed.
+- `consumererror.NewHTTPStatus` for errors resulting from an HTTP call with an error status code.
+- `consumererror.NewGRPCStatus` for errors resulting from a gRPC call with an error status code.
```

Comment:
```suggestion
- `consumererror.NewHTTPError` for errors resulting from an HTTP call with an error status code.
- `consumererror.NewGRPCError` for errors resulting from a gRPC call with an error status code.
```

Same in other places.

Should we make it even shorter `consumererror.New[GRPC|HTTP]`?

Reply:
My bad, the error names already ended in `Error`. Either way, I like the shorter names: they don't stutter and match the other error names.

Reply:
Sounds good!